src/components/application_manager/src/commands/mobile/subscribe_vehicle_data_request.cc

```diff
@@ -363,18 +363,36 @@ void SubscribeVehicleDataRequest::AddAlreadySubscribedVI(
   using namespace mobile_apis;
   VehicleInfoSubscriptions::const_iterator it_same_app =
       vi_already_subscribed_by_this_app_.begin();
+  const VehicleData& vehicle_data = MessageHelper::vehicle_data();
+
   for (; vi_already_subscribed_by_this_app_.end() != it_same_app;
        ++it_same_app) {
-    msg_params[*it_same_app][strings::result_code] =
-        VehicleDataResultCode::VDRC_DATA_ALREADY_SUBSCRIBED;
+    for (VehicleData::const_iterator vd_it = vehicle_data.begin();
```

Comment:
@LevchenkoS  this cycle looks like `find` algorithm.

Could you substitute raw loop with standard algorithm?